straxen/common.py

```diff
@@ -243,7 +247,185 @@ def get_livetime_sec(context, run_id, things=None):
         else:
             return (md['end'] - md['start']).total_seconds()
 
+@export
+def remap_channels(data, verbose=True, safe_copy=False, _tqdm=False, ):
+    """
+    There were some errors in the channel mapping of old data as described in
+        https://xe1t-wiki.lngs.infn.it/doku.php?id=xenon:xenonnt:dsg:daq:sector_swap
+        using this function, we can convert old data to reflect the right channel map
+        while loading the data. We convert both the field 'channel' as well as anything
+        that is an array of the same length of the number of channels.
+
+    :param data: numpy array of pandas dataframe
+
+    :param verbose: print messages while converting data
+
+    :param safe_copy: if True make a copy of the data prior to performing manipulations.
+        Will prevent overwrites of the internal references but does require more memory.
+    
+    :param _tqdm: bool (try to) add a tqdm wrapper to show the progress
+
+    :return: Correctly mapped data
+    """
+    # This map shows which channels were recabled. We now have to do the same in software
+    # for old runs.
+    remap = get_resource(
+        aux_repo + '/ecb6da7bd4deb98cd0a4e83b3da81c1e67505b16/remapped_channels_since_20200729_17.20UTC.csv',
+        fmt='csv')
+
+    def wr_tqdm(x):
+        """Wrap input x with tqdm"""
+        if _tqdm:
+            try:
+                return tqdm.tqdm_notebook(x)
+            except (AttributeError, ModuleNotFoundError, ImportError):
+                # ok, sorry lets not wrap but return x
+                pass
+        return x
+
+    def get_dtypes(_data):
```

Comment:
Might be also helpful for other people, maybe put it into common.py as a regular function idl.

Reply:
Done